app-i18n/uchardet/uchardet-9999.ebuild

```diff
@@ -7,10 +7,7 @@ inherit cmake-utils git-r3
 
 DESCRIPTION="An encoding detector library"
 HOMEPAGE="https://www.freedesktop.org/wiki/Software/uchardet/"
-EGIT_REPO_URI=(
-	https://anongit.freedesktop.org/git/${PN}/${PN}.git
-	git://anongit.freedesktop.org/${PN}/${PN}
-)
+EGIT_REPO_URI="https://anongit.freedesktop.org/git/${PN}/${PN}.git"
```

Comment:
HTTPS blocking is actually the reason why unsafe fallback options are a bad idea for the majority of users who aren't subject to HTTPS blocking. A fallback option of git:// (port 9418) just means an attacker needs to block HTTPS connections (port 443), forcing git clients to always revert to using a protocol which can be attacked.

Security matters aside, I'd also be surprised if port 443 was blocked more frequently than 9418 (git transport). For this reason, HTTPS is probably the most likely protocol of the two to be allowed through on the first attempt.

I agree that it's not necessarily ideal that git[curl] is required over just git[], but then again, a large majority of packages in the tree are only available over HTTPS connections. Also have a look at https://repology.org/repository/gentoo/problems which lists a significant number of ebuilds where the HOMEPAGE (and therefore probably SRC_URI as well) are permanently redirecting to a HTTPS site. I was planning on fixing those ebuilds too.

Some solutions to support your arguments, as well as the secure protocol only argument are:

1. There is the `${PN}_LIVE_REPO` variable which could be defined by users on a per-package basis to override the default protocol of HTTPS.
2. Git could also be configured on the user's Gentoo system with `git config --global url."git://".insteadOf https://` to convert all https:// URLs to git://. Rules for specific domains can also be configured this way using `git config --global url."https://github".insteadOf git://github`.
3. Modify function `git-r3_fetch()` within git-r3.eclass to perform a similar substitution to option (2) above, with configuration of substitution rules defined in make.conf.
4. Modify function `git-r3_fetch()` within git-r3.eclass to factor in new configuration options from make.conf to specify a priority for protocols to use with git, and how to handle fallback behaviour. For example, the default might be "https > ssh > git" as the priority, with fallback behaviour configured to "only use unsecured protocols if the ebuild doesn't specify a secure protocol". Another user with HTTPS blocked might configure the priority as "git" with fallback behaviour of "use any fallbacks available" (also implied by only setting the priority as "git").